lib/SensorFusion/src/ExtendedKalmanFilter.h

```diff
@@ -79,23 +90,91 @@ class ExtendedKalmanFilter : public LinearKalmanFilter
-    Eigen::MatrixXf m_covariance; /**< Covariance Matrix of the state */
+    /** Estimated state vector [positionX, positionY, velocity, orientation]^T */
+    Eigen::Vector<float, NUMBER_OF_STATES_N> m_stateVector;
+
+    /** Covariance Matrix of the state */
+    Eigen::Matrix<float, NUMBER_OF_STATES_N, NUMBER_OF_STATES_N> m_covarianceMatrix;
+
+    /** Control Input Vector u=[accelerationX, accelerationY, turnRate]^T */
+    Eigen::Vector<float, NUMBER_OF_CONTROL_INPUTS_L> m_controlInputVector;
+
+    /** Measurement Vector z=[positionOdometryX, positionOdometryY, orientationOdometry]^T */
+    Eigen::Vector<float, NUMBER_OF_MEASUREMENTS_M> m_measurementVector;
+
+    /** Index of Position in x-direction in the state vector x. */
+    static const uint8_t IDX_POSITION_X_STATE_VECTOR = 0U;
+
+    /** Index of Position in y-direction in the state vector x. */
+    static const uint8_t IDX_POSITION_Y_STATE_VECTOR = 1U;
+
+    /** Index of the velocity in the state vector x. */
+    static const uint8_t IDX_VELOCITY_STATE_VECTOR = 2U;
+
+    /** Index of the orientation in the state vector x. */
+    static const uint8_t IDX_ORIENTATION_STATE_VECTOR = 3U;
+
+    /** Index of Position in x-direction of the odometry in the measurement vector z. */
+    static const uint8_t IDX_ODOMETRY_X_MEASUREMENT_VECTOR = 0U;
+
+    /** Index of Position in y-direction of the odometry in the measurement vector z. */
+    static const uint8_t IDX_ODOMETRY_Y_MEASUREMENT_VECTOR = 1U;
+
+    /** Index of Orientation of the odometry in the measurement vector z. */
+    static const uint8_t IDX_ODOMETRY_ORIENTATION_MEASUREMENT_VECTOR = 2U;
+
+    /** Index of Acceleration in x-direction in the control input vector u. */
+    static const uint8_t IDX_ACCELERATION_X_CONTROL_INPUT_VECTOR = 0U;
+
+    /** Index of Acceleration in y-direction in the control input vector u. */
+    static const uint8_t IDX_ACCELERATION_Y_CONTROL_INPUT_VECTOR = 1U;
+
+    /** Index of Turn Rate in the control input vector u. */
+    static const uint8_t IDX_TURNRATE_CONTROL_INPUT_VECTOR = 2U;
+
+    /** Initial covariance matrix (notation in literature: P). */
+    static const Eigen::Matrix<float, NUMBER_OF_STATES_N, NUMBER_OF_STATES_N> INITIAL_COVARIANCE_MATRIX_P;
+
+    /** The covariance matrix of the process noise Matrix (notation in literature: Q). */
+    static const Eigen::Matrix<float, NUMBER_OF_STATES_N, NUMBER_OF_STATES_N> PROCESS_COVARIANCE_MATRIX_Q;
+
+    /** The observation model Matrix (notation in literature: H). */
+    static const Eigen::Matrix<float, NUMBER_OF_MEASUREMENTS_M, NUMBER_OF_STATES_N> OBSERVATION_MATRIX_H;
+
+    /** The covariance of the observation noise Matrix (notation in literature: R). */
+    static const Eigen::Matrix<float, NUMBER_OF_MEASUREMENTS_M, NUMBER_OF_MEASUREMENTS_M> OBSERVATION_NOISE_MATRIX_R;
+
+    /**
+     * Writes data from a KalmanParameter Struct into the Measurement Vector as a member variable m_measurementVector

```

Comment:
The description does not match the naming of the method: "generate..." implies that something *new* is created, the description, however, implies that something existing is just updated.

Maybe you change the name of the method or extend the description to better clarify.